File: Backend/src/main/java/hadi/springSecurity/models/embeddables/Name.java
package hadi.springSecurity.models.embeddables;

import javax.persistence.Embeddable;
import javax.validation.constraints.Size;

@Embeddable
public class Name
{
	@Size(min = 2, max = 16)
	private String firstName;
	@Size(min = 0, max = 16)
	private String middleName;
	@Size(min = 2, max = 16)
	private String lastName;
	
	
	
	public Name() {	}


	public Name(String firstName, String lastName)
	{
		this.firstName = firstName;
		this.middleName = "";
		this.lastName = lastName;
	}


	public Name(String firstName, String middleName, String lastName)
	{
		this.firstName = firstName;
		this.middleName = middleName;
		this.lastName = lastName;
	}
	
	
	public String getFirstName()
	{
		return firstName;
	}
	public void setFirstName(String firstName)
	{
		this.firstName = firstName;
	}
	public String getMiddleName()
	{
		return middleName;
	}
	public void setMiddleName(String middleName)
	{
		this.middleName = middleName;
	}
	public String getLastName()
	{
		return lastName;
	}
	public void setLastName(String lastName)
	{
		this.lastName = lastName;
	}


	@Override
	public int hashCode()
	{
		final int prime = 31;
		int result = 1;
		result = prime * result + ((firstName == null) ? 0 : firstName.hashCode());
		result = prime * result + ((lastName == null) ? 0 : lastName.hashCode());
		result = prime * result + ((middleName == null) ? 0 : middleName.hashCode());
		return result;
	}


	@Override
	public boolean equals(Object obj)
	{
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Name other = (Name) obj;
		if (firstName == null)
		{
			if (other.firstName != null)
				return false;
		} else if (!firstName.equals(other.firstName))
			return false;
		if (lastName == null)
		{
			if (other.lastName != null)
				return false;
		} else if (!lastName.equals(other.lastName))
			return false;
		if (middleName == null)
		{
			if (other.middleName != null)
				return false;
		} else if (!middleName.equals(other.middleName))
			return false;
		return true;
	}


	@Override
	public String toString()
	{
		return firstName + (middleName.equals("") ? "" : " " + middleName) + " " + lastName;
	}
	
	
	
}

File: Frontend/src/app/models/other/Name.ts
export class Name
{
    private _firstName: string;
    private _lastName: string;
    private _middleName: string;

    constructor(firstName: string, lastName: string, middleName?: string)
    {
        this._firstName = firstName;
        this._lastName = lastName;
        if (middleName != null && middleName != undefined)
        {
            this._middleName = middleName;
        }
        else
        {
            this._middleName = "";
        }
    }

    public get firstName(): string
    {
        return this._firstName;
    }

    public set firstName(value: string)
    {
        this._firstName = value;
    }

    public get lastName(): string
    {
        return this._lastName;
    }

    public set lastName(value: string)
    {
        this._lastName = value;
    }

    public get middleName(): string
    {
        return this._middleName;
    }

    public set middleName(value: string)
    {
        this._middleName = value;
    }

    public toString():string
    {
        return this.firstName + " " + ((this.middleName!= undefined)? " " + this.middleName + " " : "") + this.lastName;
    }
}